service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -620,23 +620,20 @@ public ResponseEntity<ApiCloneWorkspaceResult> cloneWorkspace(
           samService.getUserEmailFromSamAndRethrowOnInterrupt(getAuthenticatedInfo()));
     }
 
-    // TODO: PF-2694 REST API part
-    //  When we make the REST API changes, the spend profile will come with the source cloud context
-    //  and we will take it from there. Only if missing, will we take it from the workspace. For
-    //  this part, we do the retrieval and permission check using the workspace spend profile.
-    Optional<SpendProfileId> spendProfileIdOptional =
-        Optional.ofNullable(body.getSpendProfile()).map(SpendProfileId::new);
+    // Use the requested spend profile if present. If not present, use the source workspace's
+    // spend profile. If source workspace also has no spend profile, the clone operation will
+    // proceed and the destination workspace will have no spend profile.
+    SpendProfileId workspaceSpendProfileId = sourceWorkspace.getSpendProfileId().orElse(null);
     SpendProfileId spendProfileId =
-        spendProfileIdOptional.orElse(
-            sourceWorkspace
-                .getSpendProfileId()
-                .orElseThrow(
-                    () ->
-                        MissingSpendProfileException.forWorkspace(sourceWorkspace.workspaceId())));
-
-    SpendProfile spendProfile =
-        spendProfileService.authorizeLinking(
-            spendProfileId, features.isBpmGcpEnabled(), petRequest);
+        Optional.ofNullable(body.getSpendProfile())
+            .map(SpendProfileId::new)
```

Comment:
I'm having trouble understanding this-- what is the .map operation doing?

Reply:
If there's a spend profile passed in the body of the request, this will create a `SpendProfileId` from the string value. If there's no value passed in the request, then this will be an empty `Optional`

Reply:
Ah, so `SpringProfileId::new` is basically a constructor call.